File: server/src/main/java/com/xcore/application/authentication/models/role/AppUserRole.java
package com.xcore.application.authentication.models.role;

import com.xcore.application.authentication.models.user.AppUser;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;

import javax.persistence.*;
import java.io.Serializable;
import java.util.List;

@Data
@Entity
@NoArgsConstructor
@AllArgsConstructor
public class AppUserRole implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column
    private Long id;

    @Column(length = 64)
    @NonNull
    private String name;

    @Column
    @NonNull
    private ERoleAccessLevel accessLevel;

    @OneToMany(mappedBy = "id")
    private List<AppUser> users;

}

File: server/src/main/scala/com/xcore/application/authentication/models/role/AppUserRole.scala
package com.xcore.application.authentication.models.role;

import javax.persistence._
import lombok.NonNull
import org.springframework.security.core.GrantedAuthority

import scala.beans.BeanProperty;

@Entity
class AppUserRole extends Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @BeanProperty
    var id: Long = _;

    @Column(length = 64, unique = true)
    @BeanProperty
    @NonNull
    var name: String = "default";

    @Column(unique = true)
    @BeanProperty
    @Enumerated(EnumType.STRING)
    @NonNull
    var accessLevel: EAppUserRoleAccessLevel = EAppUserRoleAccessLevel.ROLE_FROZEN;

    @Transient
    def getAuthorities: java.util.List[GrantedAuthority] = accessLevel.getAuthorities;

}

